tasks/nginx.yml

```diff
@@ -18,10 +18,19 @@
     src: ../templates/nginx.j2
     dest: "/etc/nginx/sites-available/{{ inventory_hostname }}"
   notify: restart nginx
+  when: ansible_distribution != 'CentOS' and ansible_distribution != 'Red Hat Enterprise Linux'
 
 - name: nginx site enabled
   file:
     src: "/etc/nginx/sites-available/{{ inventory_hostname }}"
     dest: "/etc/nginx/sites-enabled/{{ inventory_hostname }}"
     state: link
   notify: restart nginx
+  when: ansible_distribution != 'CentOS' and ansible_distribution != 'Red Hat Enterprise Linux'
+
+- name: nginx site
+  template:
+    src: ../templates/nginx.j2
+    dest: "/etc/nginx/conf.d/{{ inventory_hostname }}.conf"
```

Comment:
It looks like only the filesystem path changes, if I'm not wrong, the template used is the same.

You could use the same approach I suggested for `www_group` and `www_data`, set these variables using `set_fact` depending on `ansible_os_family`